android/src/main/kotlin/com/revenuecat/purchases_flutter/PurchasesFlutterPlugin.kt

```diff
@@ -0,0 +1,652 @@
+                setMediaSource(mediaSource, result)
+            }
+            Constants.MethodNameConstants.setCampaign -> {
+                val campaign = call.argument<String>(Constants.MethodArgsNameConstants.campaign)
+                setCampaign(campaign, result)
+            }
+            Constants.MethodNameConstants.setAdGroup -> {
+                val adGroup = call.argument<String>(Constants.MethodArgsNameConstants.adGroup)
+                setAdGroup(adGroup, result)
+            }
+            Constants.MethodNameConstants.setAd -> {
+                val ad = call.argument<String>(Constants.MethodArgsNameConstants.ad)
+                setAd(ad, result)
+            }
+            Constants.MethodNameConstants.setKeyword -> {
+                val keyword = call.argument<String>(Constants.MethodArgsNameConstants.keyword)
+                setKeyword(keyword, result)
+            }
+            Constants.MethodNameConstants.setCreative -> {
+                val creative = call.argument<String>(Constants.MethodArgsNameConstants.creative)
+                setCreative(creative, result)
+            }
+            Constants.MethodNameConstants.collectDeviceIdentifiers -> collectDeviceIdentifiers(result)
+            Constants.MethodNameConstants.canMakePayments -> {
+                val features = call.argument<List<Int>>(Constants.MethodArgsNameConstants.features)!!
+                canMakePayments(features, result)
+            }
+            Constants.MethodNameConstants.close -> close(result)
+            Constants.MethodNameConstants.setLogHandler -> setLogHandler(result)
+            Constants.MethodNameConstants.syncObserverModeAmazonPurchase -> {
+                val productID = call.argument<String>(Constants.MethodArgsNameConstants.productID)
+                val receiptID = call.argument<String>(Constants.MethodArgsNameConstants.receiptID)
+                val amazonUserID = call.argument<String>(Constants.MethodArgsNameConstants.amazonUserID)
+                val isoCurrencyCode = call.argument<String>(Constants.MethodArgsNameConstants.isoCurrencyCode)
+                val price = call.argument<Double>(Constants.MethodArgsNameConstants.price)
+                syncObserverModeAmazonPurchase(
+                    productID, receiptID, amazonUserID, isoCurrencyCode,
+                    price, result
+                )
+            }
+            else -> result.notImplemented()
+        }
+    }
+
+    private fun setupPurchases(
+        apiKey: String?, appUserID: String?, observerMode: Boolean?,
+        useAmazon: Boolean?, result: MethodChannel.Result
+    ) {
+        if (applicationContext != null) {
+            val platformInfo = PlatformInfo(PLATFORM_NAME, PLUGIN_VERSION)
+            var store = Store.PLAY_STORE
+            if (useAmazon != null && useAmazon) {
+                store = Store.AMAZON
+            }
+            configure(
+                applicationContext!!, apiKey!!, appUserID, observerMode,
```

Comment:
Instead of crashing here, it would be better to try to handle this gracefully by calling `result.error` (I understand it would have crashes with the Java version as well, but we can improve that now)